### PR TITLE
fix(iam/group_membership): check whether the user exists before remove it from the group

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes NONE

**Special notes for your reviewer**:

**Release note**:

```release-note
check whether the user exists before remove it from the group
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccIdentityV3GroupMembership_basic'==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccIdentityV3GroupMembership_basic -timeout 360m -parallel 4
=== RUN   TestAccIdentityV3GroupMembership_basic
=== PAUSE TestAccIdentityV3GroupMembership_basic
=== CONT  TestAccIdentityV3GroupMembership_basic
--- PASS: TestAccIdentityV3GroupMembership_basic (73.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       73.062s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_identity_group_membership.go [hunk 1]
```diff
@@ -147,16 +153,24 @@ func resourceIdentityGroupMembershipV3Delete(d *schema.ResourceData, meta interf
...
+			return fmtp.Errorf("Error add user %s to group %s: %s ", u, group, r)
 		}
 	}
 	return nil
 }
 
-func removeUsersFromGroup(identityClient *golangsdk.ServiceClient, group string, userList []string) error {
+func removeUsersFromGroup(identityClient, iamClient *golangsdk.ServiceClient, group string, userList []string) error {
 	for _, u := range userList {
+		_, err := iamUsers.Get(iamClient, u).Extract()
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_identity_group_membership.go]: we can also get user details with identityClient, please see [here](https://support.huaweicloud.com/intl/en-us/api-iam/iam_08_0002.html).

Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_identity_group_membership.go]: found a better way, please have a look

**Comment by Jason-Zhang9309** [comment]
latest test result:
```
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccIdentityV3GroupMembership_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccIdentityV3GroupMembership_basic -timeout 360m -parallel 4
=== RUN   TestAccIdentityV3GroupMembership_basic
=== PAUSE TestAccIdentityV3GroupMembership_basic
=== CONT  TestAccIdentityV3GroupMembership_basic
--- PASS: TestAccIdentityV3GroupMembership_basic (74.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       74.861s
```

File: huaweicloud/resource_huaweicloud_identity_group_membership.go [hunk 2]
```diff
@@ -65,12 +67,7 @@ func resourceIdentityGroupMembershipV3Read(d *schema.ResourceData, meta interfac
 
 	allPages, err := users.ListInGroup(identityClient, group, users.ListOpts{}).AllPages()
 	if err != nil {
-		if _, b := err.(golangsdk.ErrDefault404); b {
-			d.SetId("")
-			return nil
-		} else {
-			return fmtp.Errorf("Unable to query groups: %s", err)
-		}
+		return common.CheckDeleted(d, err, "Unable to query groups")
```
Comment (niuzhenguo) [inline huaweicloud/resource_huaweicloud_identity_group_membership.go:70]: CheckDeleted will wrap the 404 error message with your specifed one ("%s: %s", your_msg_above, err), so here better to use something lie `identity_group_memeber` to specify which resource logs this error.

**Pull request merged**